db/src/lib.rs

```diff
@@ -1,6 +1,51 @@
 pub use sqlx;
 use sqlx::postgres::PgPoolOptions;
 pub use sqlx::PgPool;
+use color_eyre::Result;
+use uuid::Uuid;
+
+
+#[derive(sqlx::FromRow)]
+pub struct User {
+    pub user_id: Uuid,
+    pub user_name: String,
+}
+
+
+pub async fn create_user(pool: &PgPool, username: &str) -> color_eyre::Result<Uuid> {
```

Comment:
Oh this `clippy` is interesting 🤔 

I'll likely push a commit to turn this off, at least for now!

The clippy is saying

> docs for function which may panic missing `# Panics` section

The idea here is that in Rust there are kinda two ways to do errors. The 'right' way with returning a `Result` like we are going, or you can 'panic' with `unwrap` or `panic!` or something like that.
The lint wants you to add a comment above this function that explains why this function might panic as a way to let other developers know a panic is possible in this function

BUTT I don't _see_ what could panic here.. Maybe `query!` can panic or something... But I don't think we want/need to care about that panic here, so probably makes sense to ignore this one

I'll push a commit to this PR in like 5 min to do that!

Reply:
Ohhb 🤔  is `fetch_one` what is causing the panic here? Ahh ya I think it is!

Ok so maybe we do want to deal with this! I think we want to look at the times you used `fetch_one` and see if it makes sense to use `fetch_one` and write a panic comment or if we should switch to using `fetch_optional` (which returns an Option<T> instead of a T)

In this case I think the `fetch_one` is great, because I don't think there is any reasonable case where this query wouldn't return any results (since its creating the record). We may get an error about the username existing, but I think that will come as a different `Err` anyways, so I think panicing here is fine!

So lets add a `doc comment` https://doc.rust-lang.org/reference/comments.html#doc-comments
Above this function with a `# Panics` section, that says this function could panic if the `INSERT` row doesn't return a row, but we don't expect that to happen in any normal circumstances.

And then with that `clippy` should be happy here!

Reply:
should I comment this?

Reply:
Oh sorry I didn't see this Github Comment but ya I love this! 💯